Plugin Management: show neutral state for already up-to-date plugins when clicked on update

yashwin opened the pull request:
#### Proposed Changes

The PR adds a neutral state to the plugin update status that shows `Plugin already up to date` instead of showing `Update successful`.

#### Testing Instructions

> **_NOTE:_** You can test these changes in Calypso Blue or Green.

**Prerequisites**

Since this change is made specifically for agencies, you must set yourself(partner) as an agency if you want to test in Calypso Green - 2c49b-pb. Make sure to switch it back to the previous type. 

**Instructions**

1. Run `git checkout update/add-neutral-state-for-plugin-update` and `yarn start-jetpack-cloud` or `yarn start` if you want to test these changes in Calypso Blue.
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard -> Click on **Plugins** in the sidebar or visit http://calypso.localhost:3000/plugins/manage if you want to test these changes in Calypso Blue.
3. Click on `Edit All` -> Click on `Update Plugins` and verify the status for already up-to-date plugins shows `Plugin already up to date`

<table>
<tr>
<th>
Before
</th>
<th>
After
</th>
</tr>
<tr>
<td>
<img width="577" alt="Screenshot 2022-10-05 at 10 22 00 AM" src="https://user-images.githubusercontent.com/10586875/193983848-3b224018-11eb-4144-9792-79340b62ff5f.png">
</td>
<td>
<img width="577" alt="Screenshot 2022-10-05 at 10 21 23 AM" src="https://user-images.githubusercontent.com/10586875/193983837-257e2e32-6889-473c-af1e-51e31abd2079.png">
</td>
</tr>
</table>

#### Pre-merge Checklist

Complete applicable items on this checklist **before** merging into the trunk. Inapplicable items can be left unchecked.

The PR author and reviewer are responsible for completing the checklist.

- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] ~~Have you tested the feature in [Simple](P9HQHe-k8-p2-p2), [Atomic](P9HQHe-jW-p2-p2), and [self-hosted Jetpack sites](PCYsg-g6b-p2-p2)?~~
- [x] Have you checked for TypeScript, React, or other console errors?
- [x] Are we memoizing when appropriate (for expensive computations)? More info in [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md) and [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors)
- [x] Have we sent any new strings [for translation](PCYsg-1vr-p2-p2) ASAP?

Related to 1202619025189113-as-1203011720281491

yashwin committed:
Show neutral state for already up-to date plugins when clicked on update

rcanepa commented on client/my-sites/plugins/plugin-management-v2/plugin-action-status/style.scss:
```diff
@@ -26,7 +26,8 @@
 }
 
-.plugin-action-status-inProgress {
+.plugin-action-status-inProgress,
+.plugin-action-status-upToDate {
```
I don't think we use uppercase on class names. I recommend using `-inprogress` and `-uptodate` to follow the conventions.

yashwin replied:
I have changed `upToDate` to `up-to-date`. `inProgress` status is used in quite a lot of places, so leaving it as it is for now.

rcanepa commented on client/my-sites/plugins/plugin-management-v2/utils/get-plugin-action-statuses.ts:
```diff
@@ -8,6 +8,7 @@ export const getPluginActionStatuses = ( state: AppState ) => {
 	const inProgressStatuses = getPluginStatusesByType( state, 'inProgress' );
 	const completedStatuses = getPluginStatusesByType( state, 'completed' );
 	const errorStatuses = getPluginStatusesByType( state, 'error' );
+	const updtoDateStatuses = getPluginStatusesByType( state, 'upToDate' );
```
There is a typo here. It should be `upToDateStatuses` instead of `updtoDateStatuses`.

rcanepa commented on client/my-sites/plugins/plugin-management-v2/utils/get-plugin-action-statuses.ts:
```diff
@@ -8,6 +8,7 @@ export const getPluginActionStatuses = ( state: AppState ) => {
+	const updtoDateStatuses = getPluginStatusesByType( state, 'upToDate' );
 
-	return [ ...inProgressStatuses, ...completedStatuses, ...errorStatuses ];
+	return [ ...inProgressStatuses, ...completedStatuses, ...errorStatuses, ...updtoDateStatuses ];
```
Same here.

rcanepa reviewed (approved):
LGTM.

Everything works as expected. I think we can ship this once we address the comments I left.

Thanks, Yashwin!

yashwin committed:
Fix typo and use hyphen instead of camel case for status

yashwin committed:
Include up-to-date status in removePluginStatuses function call

yashwin committed:
Add test case to cover up-to-date status

yashwin commented:
Thanks for the review, @rcanepa. I have addressed your comments and made some changes. I have also added a test case here after the test case PR was merge.

rcanepa reviewed (approved):
LGTM now!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7567916

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7567916).
